core/src/storage/cache.py

```diff
@@ -1,36 +1,82 @@
 from trezor.crypto import random
 
 if False:
-    from typing import Optional
+    from typing import Optional, Dict, List
 
+_MAX_SESSIONS_COUNT = 10
+_SESSIONLESS_FLAG = 128
+
+# Traditional cache keys
 APP_COMMON_SEED = 0
-APP_COMMON_SEED_WITHOUT_PASSPHRASE = 1
-APP_CARDANO_ROOT = 2
-APP_MONERO_LIVE_REFRESH = 3
+APP_CARDANO_ROOT = 1
+APP_MONERO_LIVE_REFRESH = 2
+
+# Keys that are valid across sessions
+APP_COMMON_SEED_WITHOUT_PASSPHRASE = 1 | _SESSIONLESS_FLAG
+
 
-_cache_session_id = None  # type: Optional[bytes]
-_cache = {}
+_active_session_id = None  # type: Optional[bytes]
+_caches = {}  # type: Dict[bytes, Dict[int, Any]]
```

Comment:
nit: perhaps we should have `CacheDict = Dict[int, Any]`, and use that as the type hint